iavl/iavl_proof.go

```diff
@@ -17,20 +18,28 @@ type IAVLProof struct {
 	RootHash   []byte
 }
 
-func (proof *IAVLProof) Verify(key []byte, value []byte, root []byte) bool {
+func (proof *IAVLProof) Verify(key []byte, value []byte, root []byte, version int) bool {
```

Comment:
Cool. Does Save work if we're using a memDB? If so, great. I think it once was set to panic, so Hash() was useful.